frame/identity/src/lib.rs

```diff
@@ -451,9 +456,112 @@ decl_error! {
+		TooManyRegistrars,
 }
 }
 
+/// Functions for calcuating the weight of dispatchables.
+mod weight_for {
+	use frame_support::weights::{RuntimeDbWeight, Weight};
+
+	/// Weight calculation for `set_identity`.
+	pub(crate) fn set_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(1, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 150_000_000 // constant
+			+ 700_000 * judgements.into() // R
+			+ 3_000_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `set_subs`.
+	pub(crate) fn set_subs(
+		db: RuntimeDbWeight,
+		old_subs: impl Into<Weight> + Copy,
+		subs: impl Into<Weight> + Copy
+	) -> Weight {
+		db.reads(1) // storage-exists (`IdentityOf::contains_key`)
+			+ db.reads_writes(1, 1) // balance ops
+			+ db.reads_writes(1, old_subs.into()) // P old DB deletions
+			+ db.writes(subs.into() + 1) // S + 1 new DB writes
+			+ 130_000_000 // constant
+			+ 5_200_000 * old_subs.into() // P
+			+ 7_300_000 * subs.into() // S
+	}
+
+	/// Weight calculation for `clear_identity`.
+	pub(crate) fn clear_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		subs: impl Into<Weight> + Copy,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, subs.into() + 2) // S + 2 deletions
+			+ db.reads_writes(1, 1) // balance ops
+			+ 160_000_000 // constant
+			+ 500_000 * judgements.into() // R
+			+ 5_400_000 * subs.into() // S
+			+ 2_000_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `request_judgement`.
+	pub(crate) fn request_judgement(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 180_000_000 // constant
+			+ 950_000 * judgements.into() // R
+			+ 3_400_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `cancel_request`.
+	pub(crate) fn cancel_request(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(1, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 150_000_000 // constant
+			+ 600_000 * judgements.into() // R
+			+ 3_600_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `provide_judgement`.
+	pub(crate) fn provide_judgement(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, 1)
+			+ 120_000_000 // constant
+			+ 1_100_000 * judgements.into() // R
+			+ 3_500_000 * extra_fields.into()// X
+	}
+
+	/// Weight calculation for `kill_identity`.
+	pub(crate) fn kill_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		subs: impl Into<Weight> + Copy,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(3, subs.into() + 3) // 2 `take`s + S deletions
+			+ db.reads_writes(1, 1) // balance ops
```

Comment:
This case balance op is correct

Reply:
But you seem to be missing a write to the treasury account (which I think you would skip anyway since treasury is written to every block, but for further analysis, would be good to double check why you might have missed treasury)

Reply:
I should have noted that the I could not account for 1 read and 1 write in the first line.
You think that might be the treasury? Where does it come in? Through `Slashed::on_unbalanced`?

Reply:
Yes, you can see the configuration in the Node runtime, but also you can see it in our DB logs

Reply:
So [the benchmark](https://www.shawntabrizi.com/substrate-graph-benchmarks/db-stats.html?p=identity&e=kill_identity) says there are `4` reads  and `subs + 4` writes.
I'm happy to subtract 1 for the balances, but I don't know how to trace whether the treasury one is a duplicate. With "DB logs" do you mean the logs under the DB benchmark? How would I tell that it's the treasury? Is it a special address?

Reply:
I have added treasury to our "whitelist" so it should show up now explicitly in the name and remove itself automatically from the read count.

The way I figured it out is that i used the DB logs and walked through the code to identify each read/write operation on the DB log with some code that was written.

I suggest you do the same